pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin_test.go

```diff
@@ -0,0 +1,485 @@
+									MaxStreamDuration:     parseDuration("10m"),
+									MaxConnectionDuration: parseDuration("10m"),
+								},
+							},
+						},
+					}},
+			},
+			expectedCluster:  "basic_inbound_cluster.golden.yaml",
+			expectedListener: "basic_inbound_listener.golden.yaml",
+		}),
+		Entry("outbound with defaults when http conf missing", sidecarTestCase{
+			resources: []core_xds.Resource{{
+				Name:     "outbound",
+				Origin:   generator.OriginOutbound,
+				Resource: httpOutboundListenerWith(10001),
+			},
+				{
+					Name:     "outbound",
+					Origin:   generator.OriginOutbound,
+					Resource: clusterWithName("other-service"),
+				}},
+			toRules: core_xds.ToRules{
+				Rules: []*core_xds.Rule{
+					{
+						Subset: core_xds.Subset{
+							{
+								Key:   mesh_proto.ServiceTag,
+								Value: "other-service",
+							},
+						},
+						Conf: api.Conf{
+							ConnectionTimeout: parseDuration("10s"),
+							IdleTimeout:       parseDuration("1h"),
+						},
+					},
+				},
+			},
+			expectedCluster:  "outbound_with_defaults_cluster.golden.yaml",
+			expectedListener: "outbound_with_defaults_listener.golden.yaml",
+		}),
+	)
+
+	It("should generate proper Envoy config for MeshGateway Dataplanes",
+		func() {
+			toRules := core_xds.ToRules{
+				Rules: []*core_xds.Rule{
+					{
+						Subset: core_xds.Subset{},
+						Conf: api.Conf{
+							ConnectionTimeout: parseDuration("10s"),
+							IdleTimeout:       parseDuration("1h"),
+							Http: &api.Http{
+								RequestTimeout:        parseDuration("5s"),
+								StreamIdleTimeout:     parseDuration("1s"),
+								MaxStreamDuration:     parseDuration("10m"),
+								MaxConnectionDuration: parseDuration("10m"),
+							},
+						},
+					},
+				}}
+
+			resources := xds_context.NewResources()
+			resources.MeshLocalResources[core_mesh.MeshGatewayType] = GatewayResources()
+			resources.MeshLocalResources[core_mesh.MeshGatewayRouteType] = GatewayRoutes()
+
+			context := createSimpleMeshContextWith(resources)
+			proxy := xds.Proxy{
+				APIVersion: "v3",
+				Dataplane: &core_mesh.DataplaneResource{
+					Meta: &test_model.ResourceMeta{
+						Mesh: "default",
+						Name: "gateway",
+					},
+					Spec: &mesh_proto.Dataplane{
+						Networking: &mesh_proto.Dataplane_Networking{
+							Address: "127.0.0.1",
+							Gateway: &mesh_proto.Dataplane_Networking_Gateway{
+								Tags: map[string]string{
+									mesh_proto.ServiceTag: "gateway",
+								},
+								Type: mesh_proto.Dataplane_Networking_Gateway_BUILTIN,
+							},
+						},
+					},
+				},
+				Policies: xds.MatchedPolicies{
+					Dynamic: map[core_model.ResourceType]xds.TypedMatchingPolicies{
+						api.MeshTimeoutType: {
+							Type:    api.MeshTimeoutType,
+							ToRules: toRules,
+						},
+					},
+				},
+			}
+			gatewayGenerator := gatewayGenerator()
+			generatedResources, err := gatewayGenerator.Generate(context, &proxy)
+			if err != nil {
+				return
```

Comment:
that is strange. Why would we ignore the rest of the test if there is an error?

Reply:
👍